src/plugins/elasticsearch_ui_shared/static/forms/hook_form_lib/lib/subject.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+type Listener<T> = (value: T) => void;
+
+export interface Subscription {
+  unsubscribe: () => void;
+}
+
+export class Subject<T> {
+  private callbacks: Set<Listener<T>> = new Set();
+  value: T;
+
+  constructor(value: T) {
+    this.value = value;
+  }
+
+  subscribe(fn: Listener<T>): Subscription {
+    this.callbacks.add(fn);
+
+    fn(this.value);
```

Comment:
Might be worth popping this inside a
```
setTimeout(() => { fn(this.value) }, 0);
```
so that the JS scheduler runs it async and any post subscription/setup logic doesn't run after the sub listener.

Reply:
Great point! Thanks for the catch 👍